src/adapter/src/optimize/peek.rs

```diff
@@ -193,22 +193,11 @@ impl Optimize<HirRelationExpr> for Optimizer {
 
     fn optimize(&mut self, expr: HirRelationExpr) -> Result<Self::To, OptimizerError> {
         // Trace the pipeline input under `optimize/raw`.
-        debug_span!(target: "optimizer", "raw").in_scope(|| {
-            trace_plan(&expr);
-        });
+        trace_plan!(at: "raw", &expr);
 
         // HIR ⇒ MIR lowering and decorrelation
         let expr = expr.lower(&self.config)?;
 
-        // MIR ⇒ MIR optimization (local)
-        self.optimize(expr)
-    }
-}
-
-impl Optimize<MirRelationExpr> for Optimizer {
```

Comment:
This `impl` was needed before I unified the `sequence_` and `explain_` paths in `inner/peek.rs` as part of the off-thread optimization effort. After the second and third commits of #24581 all `peek::Optimizer` instances start with an `HirRelationExpr`.